src/pysource/sensitivity.py

```diff
@@ -40,9 +40,10 @@ def grad_expr(gradm, u, v, model, w=None, freq=None, dft_sub=None, isic=False):
     isic: Bool
         Whether or not to use inverse scattering imaging condition (not supported yet)
     """
-    ic_func = ic_dict[func_name(freq=freq, isic=isic)]
+    ic_func = ic_dict[func_name(freq=freq, ic=ic)]
     expr = ic_func(as_tuple(u), as_tuple(v), model, freq=freq, factor=dft_sub, w=w)
-    if model.fs:
+    if model.fs and ic in ["fwi", "isic"]:
+        # Only need `fs` processing for isic for the spatial derivatives.
```

Comment:
Unless the imaging condition involves spatial differential operator the free-surface condition isn't needed since it doesn't use any values above the surface